plugins/com.google.cloud.tools.eclipse.appengine.localserver.test/src/com/google/cloud/tools/eclipse/appengine/localserver/server/ModuleUtilsTest.java

```diff
@@ -96,4 +102,42 @@ private void mockAppEngineWebXml(String testfile) throws CoreException {
     Mockito.when(descriptorFile.getContents()).thenReturn(in);
   }
 
+  @Test
+  public void testGetAllModules_single() {
+    IServer server = Mockito.mock(IServer.class);
+    Mockito.when(server.getModules()).thenReturn(new IModule[] {module});
+    Mockito.when(server.getChildModules(any(IModule[].class), any(IProgressMonitor.class)))
+        .thenReturn(new IModule[0]);
+
+    IModule[] result = ModuleUtils.getAllModules(server);
+    Assert.assertNotNull(result);
+    Assert.assertEquals(1, result.length);
+    Assert.assertThat(result[0], Matchers.sameInstance(module));
+  }
+
+  @Test
+  public void testGetAllModules_multi() {
+    IServer server = Mockito.mock(IServer.class);
+    Mockito.when(server.getModules()).thenReturn(new IModule[] {module});
+    Mockito.when(server.getChildModules(any(IModule[].class), any(IProgressMonitor.class)))
+        .thenReturn(new IModule[0]);
+
+    IModule module2a = Mockito.mock(IModule.class, "module2a");
+    IModule module2b = Mockito.mock(IModule.class, "module2b");
+    Mockito.when(server.getChildModules(AdditionalMatchers.aryEq(new IModule[] {module}),
+        any(IProgressMonitor.class))).thenReturn(new IModule[] {module2a, module2b});
+
+    IModule module3 = Mockito.mock(IModule.class, "module3");
+    Mockito.when(server.getChildModules(AdditionalMatchers.aryEq(new IModule[] {module, module2b}),
+        any(IProgressMonitor.class))).thenReturn(new IModule[] {module3});
+
+    IModule[] result = ModuleUtils.getAllModules(server);
+    Assert.assertNotNull(result);
+    Assert.assertEquals(4, result.length);
+    Assert.assertThat(result[0], Matchers.sameInstance(module));
```

Comment:
You could use `org.hamcrest.collection.IsArrayContainingInAnyOrder` or `org.hamcrest.collection.IsArrayContainingInOrder` instead of checking one-by-one